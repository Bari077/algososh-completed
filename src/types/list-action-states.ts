export enum ActionStates {
    AddInHead = "AddInHead",
    AddInTail = "AddInTail",
    AddByIndex = "AddByIndex",
    DeleteFromHead = "DeleteFromHead",
    DeleteFromTail = "DeleteFromTail",
    DeleteByIndex = "DeleteByIndex",
  }