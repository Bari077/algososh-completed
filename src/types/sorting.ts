import { ElementStates } from "./element-states";

export type TChar = {
    item: string | number;
    id: number;
    state?: ElementStates
  }