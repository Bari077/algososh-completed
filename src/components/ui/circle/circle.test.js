import React from "react";
import renderer from 'react-test-renderer';
import { ElementStates } from "../../../types/element-states";

import { Circle } from "./circle";

describe('Circle', ()=> {
    it('should render whithout letter', ()=> {
        const tree = renderer.create(<Circle />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith letter', ()=> {
        const tree = renderer.create(<Circle letter="a"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith head', ()=> {
        const tree = renderer.create(<Circle head="test"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith react-element in head', ()=> {
        const tree = renderer.create(<Circle head={<Circle />}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith tail', ()=> {
        const tree = renderer.create(<Circle tail="test"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith react-element in tail', ()=> {
        const tree = renderer.create(<Circle tail={<Circle />}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith index', ()=> {
        const tree = renderer.create(<Circle index={1}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith prop isSmall === true', ()=> {
        const tree = renderer.create(<Circle isSmall={true}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith default state', ()=> {
        const tree = renderer.create(<Circle state={ElementStates.Default}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith changing state', ()=> {
        const tree = renderer.create(<Circle state={ElementStates.Changing}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render whith modified state', ()=> {
        const tree = renderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });


})