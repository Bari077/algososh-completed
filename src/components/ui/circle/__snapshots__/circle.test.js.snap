// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle should render whith changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render whith default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render whith head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    test
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render whith index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle should render whith letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      a
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render whith modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render whith prop isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render whith react-element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render whith react-element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle should render whith tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test
  </div>
</div>
`;

exports[`Circle should render whithout letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
